Cargo.toml

```diff
@@ -63,6 +63,7 @@ directories = "6"
 displaydoc = "0.2"
 educe = { version = "0.6.0", features = ["Debug"], default-features = false }
 either = "1"
+enumflags2 = "0.7"
```

Comment:
Is using the `bitflags` crate instead feasible? It's already in the dependency tree.

Reply:
@LesnyRumcajs `bitflags` support `struct` instead of `enum` which is not the best for our use case, it's either unfeasible or requiring a bigger refactor, I'd prefer doing that in a subsequent PR. What do you think?